postreise/plot/plot_pie_generation_vs_capacity.py

```diff
@@ -0,0 +1,248 @@
+    resource_colors=None,
+    min_percentage=0,
+):
+    """Plot any number of scenarios as pie charts with two columns per scenario -
+    generation and capacity.
+
+    :param list/str areas: list of area(s), each area is one of *loadzone*, *state*,
+        *state abbreviation*, *interconnect*, *'all'*
+    :param list/str area_types: list of area_type(s), each area_type is one of
+        *'loadzone'*, *'state'*, *'state_abbr'*, *'interconnect'*, defaults to None.
+    :param int/list/str scenario_ids: list of scenario id(s), defaults to None.
+    :param list/str scenario_names: list of scenario name(s) of same len as scenario
+        ids, defaults to None
+    :param tuple time_range: [start_timestamp, end_timestamp] where each time stamp
+        is pandas.Timestamp/numpy.datetime64/datetime.datetime. If None, the entire
+        time range is used for the given scenario.
+    :param str time_zone: new time zone, defaults to None, which uses UTC.
+    :param list custom_data: list of dictionaries with each element being
+        hand-generated data as returned by
+        :func:`postreise.plot_bar_generation_vs_capacity.make_gen_cap_custom_data`,
+        defaults to None.
+    :param dict resource_labels: a dictionary with keys being resource types and values
+        being labels, which uses to customize resource labels for selected resource
+        types to show in the plots. Defaults to None, in which case a default set of
+        labels is used.
+    :param dict resource_colors: a dictionary with keys being resource types and values
+        being colors, which uses to customize resource colors for selected resource
+        types to show in the plots. Defaults to None, in which case a default set of
+        colors is used.
+    :param float min_percentage: roll up small pie pieces into a single category,
+        resources with percentage less than the set value will be pooled together,
+        defaults to 0.
+    :raises ValueError:
+        if area_types are provided in a different length of areas and/or
+        if scenario_names are provided in a different length of scenario_ids and/or
+        if less than two scenario_ids and/or custom_data in total is provided and/or
+    :raises TypeError:
+        if resource_labels are provided but not in a dictionary format and/or
+        if resource_colors are provided but not in a dictionary format and/or
+
+    .. note::
+        if one wants to plot scenario data and custom data together, custom data MUST be
+        in TWh for generation and GW for capacity in order to conduct appropriate
+        comparison.
+    """
+    if isinstance(areas, str):
+        areas = [areas]
+    if isinstance(area_types, str):
+        area_types = [area_types]
+    if not area_types:
+        area_types = [None] * len(areas)
+    if len(areas) != len(area_types):
+        raise ValueError(
+            "ERROR: if area_types are provided, it should have the same number of entries with areas."
```

Comment:
ERROR: if area_types are provided, it must have the same number of entries than areas.

Reply:
Done.